components/contract-components/released-contract/markdown-renderer.tsx

```diff
@@ -11,29 +11,76 @@ import {
   UnorderedList,
   chakra,
 } from "@chakra-ui/react";
+import { onlyText } from "react-children-utilities";
 import ReactMarkdown from "react-markdown";
 import remarkGfm from "remark-gfm";
-import { Card, Heading, Text } from "tw-components";
+import { Card, CodeBlock, Heading, Text } from "tw-components";
 
 const ChakraReactMarkdown = chakra(ReactMarkdown);
 
 export const MarkdownRenderer: React.FC<
   BoxProps & { markdownText: string }
 > = ({ markdownText, ...restProps }) => {
+  const commonHeadingProps = {
+    lineHeight: "1.25",
+    mb: 4,
+    pb: 2,
+  };
+
   return (
     <ChakraReactMarkdown
       {...restProps}
       remarkPlugins={[remarkGfm]}
       components={{
-        h1: (props) => <Heading size="title.lg" {...props} />,
-        h2: (props) => <Heading size="title.md" {...props} />,
-        h3: (props) => <Heading size="title.sm" {...props} />,
-        h4: (props) => <Heading size="subtitle.md" {...props} />,
-        h5: (props) => <Heading size="subtitle.sm" {...props} />,
-        h6: (props) => <Heading size="label.md" {...props} />,
-        p: (props) => <Text size="body.md" {...props} />,
+        h1: (props) => (
+          <Heading
+            size="title.lg"
+            borderBottom="1px solid"
+            borderBottomColor="borderColor"
+            {...commonHeadingProps}
+            {...props}
+          />
+        ),
+        h2: (props) => (
+          <Heading
+            size="title.md"
+            borderBottom="1px solid"
+            borderBottomColor="borderColor"
+            mt={6}
+            {...commonHeadingProps}
+            {...props}
+          />
+        ),
+        h3: (props) => (
+          <Heading size="title.sm" {...commonHeadingProps} {...props} />
+        ),
+        h4: (props) => (
+          <Heading size="subtitle.md" {...commonHeadingProps} {...props} />
+        ),
+        h5: (props) => (
+          <Heading size="subtitle.sm" {...commonHeadingProps} {...props} />
+        ),
+        h6: (props) => (
+          <Heading size="label.md" {...commonHeadingProps} {...props} />
+        ),
+        pre: (props) => (
+          <CodeBlock
+            code={onlyText(props.children).trim()}
+            language="jsx"
```

Comment:
nah, we need to get this from the props if we can, can definitely not be generic `JSX`

Reply:
this might be available from props if it's ```jsx style ? have to dig into it and see if it is available.